extensions/wikia/Wall/WallExternalController.class.php

```diff
@@ -278,6 +277,13 @@ public function postNewMessage() {
 		} catch ( WallBuilderException $builderException ) {
 			\Wikia\Logger\WikiaLogger::instance()->error( $builderException->getMessage(), $builderException->getContext() );
 			$this->response->setVal( 'status', false );
+			$this->response->setVal( 'reason', 'other' );
+			$this->response->setCode( WikiaResponse::RESPONSE_CODE_INTERNAL_SERVER_ERROR );
+			return;
+		} catch ( InappropriateContentException $exception) {
+			\Wikia\Logger\WikiaLogger::instance()->error( $exception->getMessage(), $exception->getContext() );
+			$this->response->setVal( 'status', false );
+			$this->response->setVal( 'reason', 'badcontent' );
 			$this->response->setCode( WikiaResponse::RESPONSE_CODE_INTERNAL_SERVER_ERROR );
```

Comment:
Let's use a different response code here, to indicate user was prevented from performing the action deliberately